Assets/Scripts/Player/PlayerGhostRun.cs

```diff
@@ -213,7 +213,15 @@ private void UpdateGhost()
         {
             ghostRunner.transform.eulerAngles = Vector3.zero;
             ghostCamera.transform.eulerAngles = pastRunCameraRotationData[currentDataIndex];
-            inGameUI.currentSpeed = pastRunVelocityData[currentDataIndex];
+            if (pastRunVelocityData.Length > currentDataIndex)

```

Comment:
should be `if(pastRunVelocityData != null && pastRunVelocityData.Length >= currentDataIndex)`